### ISSUE TITLE
Labfile9

### ISSUE THREAD

**Issue by RiteshKumarNayak125** [RiteshKumarNayak125/FirstContribution#1]
Notepad

**Comment by RiteshKumarNayak125** [comment]
import numpy as np
import matplotlib.pyplot as plt

# Generate random data
np.random.seed(0)
X = 2 * np.random.rand(100, 1)
y = 4 + 3 * X.squeeze() + np.random.randn(100)

# Plot the generated data
plt.scatter(X, y)
plt.xlabel('X')
plt.ylabel('y')
plt.title('Randomly Generated Data')
plt.show()

# Perform linear regression
from sklearn.linear_model import LinearRegression

model = LinearRegression()
model.fit(X, y.reshape(-1, 1))  # Reshape y to avoid a warning

# Make predictions
X_new = np.array([[0], [2]])
y_pred = model.predict(X_new)

# Plot the linear regression line
plt.plot(X_new, y_pred, 'r-')
plt.scatter(X, y)
plt.xlabel('X')
plt.ylabel('y')
plt.title('Linear Regression')
plt.show()

# Evaluate the model
from sklearn.metrics import mean_squared_error, r2_score

y_pred_all = model.predict(X)
mse = mean_squared_error(y, y_pred_all)
r2 = r2_score(y, y_pred_all)
print("Mean Squared Error:", mse)
print("R^2 Score:", r2)
